nomenclature/region_mapping_models.py

```diff
@@ -0,0 +1,71 @@
+from pydantic import BaseModel, ValidationError, validator, root_validator
+from typing import Dict, Iterable, Optional, List, Union
+from pathlib import Path
+import yaml
+
+
+class NativeRegion(BaseModel):
+    name: str  # name of the region in the incoming data
+    rename: Optional[str]  # new name for the scenario explorer
+    # this new name can be automatically verfied against
+    # the regions defined in definitions/regions/regions.yaml
+
+
+class CommonRegion(BaseModel):
+    name: str  # can also be verfied against the defined regions
+    constituent_regions: List[NativeRegion]  # list of native regions
+
+
+class RegionAggregationMapping(BaseModel):
+    model: str  # if we have a list of allowed models we could also verify the model name here
+    # we can require that we have to have least either native_regions or
+    # aggregation_regions
+    native_regions: Optional[List[NativeRegion]]
+    common_regions: Optional[List[CommonRegion]]
+
+
+# should probably be moved inside RegionAggregationMapping as a classmethod
+def convert_region_mapping(file: Union[Path, str]) -> RegionAggregationMapping:
+    with open(file, "r") as stream:
+        mapping_input = yaml.safe_load(stream)
+    # check if we have a key called
+    # we need a model name and either native or common regions
```

Comment:
Not sure if we need to have either native-regions or common-regions. From the point of view of a scenario submission, if there is no mapping, the data is imported as is (without any renaming or aggregation).

Reply:
If there is no mapping at all will we still have a yml file? Would this then just contain the model name and nothing else?
I wrote that comment under the assumption that we only create a yml file if we either do renaming or aggregation or both.

Reply:
I'd see that as two separate questions:
 - Do we create a yaml file (e.g., as part of the model registration) if there is no renaming as part of the processing? This could be yes or no, but no need to make a decision on this now.
 - Do we require that a yaml file can **only** be valid if it has some renaming? This makes the code more complicated and limits our options for the first question later on, when we implement a model registration workflow. So I'd say that we do **not require** this...

Reply:
Ok, I've updated the PR. Now I allow only a model and there's a test for that.